decode.go

```diff
@@ -308,7 +308,7 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 		// Decode into the object the interface points to.
 		// XXX perhaps better ONLY to support self-decoding
 		// for interface fields?
-		return de.putvalue(wiretype, val.Elem(), v, vb)
+		return Decode(vb, val.Interface())
```

Comment:
Wait, if we ignore the `wiretype` and the other arguments here it should be feasible (and easier to grasp) to call `Decode` on `val.Interface` earlier (from the appropriate callee) instead. Sorry, I still need some time to investigate how (and why this works) and if there is a problem